ichnaea/rate_limit.py

```diff
@@ -0,0 +1,12 @@
+from redis import ConnectionError
+
+
+def rate_limit(redis_client, key, maxreq=0, expire=86400):
+    if maxreq:
+        try:
+            redis_client.set(key, 0, ex=expire, nx=True)
+            count = redis_client.incr(key, 1)
+            return count > maxreq
+        except ConnectionError:  # pragma: no cover
+            # If we cannot connect to Redis, disable rate limitation.
```

Comment:
This behavior is fine for users calling us. But when we call an external party, we don't want to make any requests if we cannot figure out the current rate limit.

So this needs one more function argument, something like `fail/pass_on_error`.